server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/DSNBounce.java

```diff
@@ -88,112 +86,88 @@
-                        + ", lastUpdated: " + newMail.getLastUpdated() + ", errorMessage: " + newMail.getErrorMessage());
+                log("Processing a bounce request for a message with an empty reverse-path.  No bounce will be sent.");
             }
+            if (!getPassThrough(originalMail)) {
+                originalMail.setState(Mail.GHOST);
+            }
+            return;
+        }
 
-            // create the bounce message
-            MimeMessage newMessage = new MimeMessage(Session.getDefaultInstance(System.getProperties(), null));
-
-            MimeMultipartReport multipart = new MimeMultipartReport();
-            multipart.setReportType("delivery-status");
-
-            // part 1: descripive text message
-            MimeBodyPart part1 = createTextMsg(originalMail);
-            multipart.addBodyPart(part1);
+        MailImpl newMail = new MailImpl(originalMail);
+        try {
 
-            // part 2: DSN
-            MimeBodyPart part2 = createDSN(originalMail);
-            multipart.addBodyPart(part2);
+            newMail.setRemoteHost(getRemoteHost());
+            newMail.setRemoteAddr(getRemoteAddr());
+            newMail.setRecipients(getSenderAsList(originalMail));
 
-            // part 3: original mail (optional)
-            if (getAttachmentType() != NONE) {
-                MimeBodyPart part3 = createAttachedOriginal(originalMail, getAttachmentType());
-                multipart.addBodyPart(part3);
+            if (isDebug) {
+                log("New mail - sender: " + newMail.getSender() + ", recipients: " + arrayToString(newMail.getRecipients().toArray()) + ", name: " + newMail.getName() + ", remoteHost: " + newMail.getRemoteHost() + ", remoteAddr: " + newMail.getRemoteAddr() + ", state: " + newMail.getState()
+                        + ", lastUpdated: " + newMail.getLastUpdated() + ", errorMessage: " + newMail.getErrorMessage());
             }
 
-            // stuffing all together
-            newMessage.setContent(multipart);
-            newMessage.setHeader(RFC2822Headers.CONTENT_TYPE, multipart.getContentType());
-            newMail.setMessage(newMessage);
+            newMail.setMessage(createBounceMessage(originalMail));
 
             // Set additional headers
             setRecipients(newMail, getRecipients(originalMail), originalMail);
             setTo(newMail, getTo(originalMail), originalMail);
             setSubjectPrefix(newMail, getSubjectPrefix(originalMail), originalMail);
             if (newMail.getMessage().getHeader(RFC2822Headers.DATE) == null) {
-                newMail.getMessage().setHeader(RFC2822Headers.DATE, rfc822DateFormat.format(new Date()));
+                newMail.getMessage().setHeader(RFC2822Headers.DATE, RFC8822_DATE_FORMAT.format(new Date()));
```

Comment:
Some tests added, refactoring/testing SimplifiedDateFormat should be done in MAILET-132